### ISSUE TITLE
Interval customization per activity instance

### ISSUE THREAD

**Issue by isuru89** [isuru89/moodle-reminders-for-calendar-events#57]
Currently, activity-specific settings support below options for each activity.
  * `Enabled`
  * `Overdue enabled`

This can be extended to support a customizable schedule as well. Such as new options will look like, 
  * `Schedule`:   [x] 7 days  [x] 3 days [x] 1 day

**Note**: `Schedule` option will enable only if `Enabled` is checked. Also when an option is disabled at plugin global settings, that option will disable here too.

**Comment by isuru89** [comment]
Something similar to below.

![Reminders 3-22-2020 9-43-52 AM](https://user-images.githubusercontent.com/1499404/77242187-b3885b80-6c21-11ea-8abf-a4266d274c57.png)

**Comment by mwehr** [comment]
Hi Isuru,

First of all thanks for the exciting new features.
I just tested the new "Interval customization per activity instance" feature.
I think I may have found a litte bug. The checkboxes for the "Day Scheduler" are not pre set in case of global settings.
I think on line 121 in file coursesettings_form.php the is $mform->setDefault call missing.
```php
foreach ($daysarray as $dkey => $dvalue) {
     $trefkey = "activityglobal_$dkey";
     $daykey = "activity_".$activity->id."_$dkey";
     $activitydayarray[] = $mform->createElement('advcheckbox', $daykey, '', $dvalue);
     $mform->disabledIf($daykey, $trefkey, 'eq', 0);
     $mform->setDefault($daykey, $dvalue); // set checkmark
}
```

Maybe a "reset do defaults" per activity for the days override part would be a nice addon to his feature?

regards
Mario

**Comment by isuru89** [comment]
@mwehr You are absolutely right!!!

**Comment by isuru89** [comment]
Released with v2.1
